### PR TITLE
CT-1264 alter column handler

### PR DESCRIPTION
**Pull request by jirkasemmler**
Jira: [CT-1270](https://keboola.atlassian.net/browse/CT-1270)
Connection PR: https://github.com/keboola/connection/pull/4883/files
SAPI PR:  https://github.com/keboola/storage-api-php-client/pull/1283

storage backend command: https://github.com/keboola/storage-backend/pull/132
storage backend table utils: https://github.com/keboola/storage-backend/pull/129
BQ driver: https://github.com/keboola/php-storage-driver-bigquery/pull/117

---

potrebuje https://github.com/keboola/storage-backend/pull/129 v table utils a common driveru

[CT-1271]: https://keboola.atlassian.net/browse/CT-1271?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CT-1270]: https://keboola.atlassian.net/browse/CT-1270?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by romanbracinik** [comment]
👁️

File: src/Handler/Table/Alter/AlterColumnHandler.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+        $bqClient = $this->clientManager->getBigQueryClient($runtimeOptions->getRunId(), $credentials);
+
+        // define columns
+        // validate
+        assert($columnDefinition->getName() !== '', 'TableColumnShared.name is required');
+        assert($columnDefinition->getType() !== '', 'TableColumnShared.type is required');
+
+        $bqColumn =
+            new Bigquery($columnDefinition->getType(), [
```
Comment (romanbracinik) [inline src/Handler/Table/Alter/AlterColumnHandler.php:64]: to neviem ci je schvalne :D

File: tests/functional/BaseCase.php [hunk 2]
```diff
@@ -26,27 +27,32 @@
...
 use PHPUnitRetry\RetryTrait;
+use Psr\Log\LogLevel;
 use Symfony\Component\Lock\LockFactory;
 use Symfony\Component\Lock\Store\FlockStore;
 use Throwable;
 
 class BaseCase extends TestCase
 {
-    use RetryTrait;
+//    use RetryTrait;
```
Comment (romanbracinik) [inline tests/functional/BaseCase.php]: ty jo mali by sme fixnut aby to loklane nerobilo retry

Comment (romanbracinik) [reply tests/functional/BaseCase.php]: https://keboola.atlassian.net/browse/CT-1306

**Review by romanbracinik — APPROVED** [review]
lgtm, este som odkomentoval tu traitu

File: src/Handler/Table/Alter/AlterColumnHandler.php (hunk 1 above)
Comment (jirkasemmler) [reply src/Handler/Table/Alter/AlterColumnHandler.php:64]: nevim co myslis - ten code style?

File: tests/functional/BaseCase.php (hunk 2 above)
Comment (jirkasemmler) [reply tests/functional/BaseCase.php]: omg.. good catch

File: src/Handler/Table/Alter/AlterColumnHandler.php (hunk 1 above)
Comment (romanbracinik) [reply src/Handler/Table/Alter/AlterColumnHandler.php:64]: jj jak je novy riadok po tom =

File: src/Handler/Table/Alter/AlterColumnHandler.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            $metadataKeysToUpdate = iterator_to_array($command->getAttributesToUpdate()->getIterator());
+            $alterColumnCommands = $builder->getUpdateColumnFromDefinitionQuery(
+                $bqColumn,
+                $databaseName,
+                $command->getTableName(),
+                $columnDefinition->getName(),
+                $metadataKeysToUpdate,
+            );
+        } catch (QueryBuilderException $e) {
+            throw new AlterColumnException($e->getMessage(), 0, $e);
```
Comment (zajca) [inline src/Handler/Table/Alter/AlterColumnHandler.php:86]: proč je tu sakra code: 0?

**Pull request merged**